mlflow/models/utils.py

```diff
@@ -489,15 +489,29 @@ def _reshape_and_cast_pandas_column_values(name, pd_series, tensor_spec):
                 tensor_spec.type
             )
         except ValueError:
-            raise MlflowException(err_msg, error_code=INVALID_PARAMETER_VALUE)
+            raise MlflowException(reshape_err_msg, error_code=INVALID_PARAMETER_VALUE)
         if len(reshaped_numpy_arr) != len(pd_series):
-            raise MlflowException(err_msg, error_code=INVALID_PARAMETER_VALUE)
+            raise MlflowException(reshape_err_msg, error_code=INVALID_PARAMETER_VALUE)
+        return reshaped_numpy_arr
+    elif isinstance(pd_series[0], np.ndarray):
+        reshape_err_msg = (
+            f"The value in the Input DataFrame column '{name}' could not be converted to the "
+            f"expected shape of: '{tensor_spec.shape}'. Ensure that each of the input numpy "
+            "array elements are of uniform length and can be reshaped to above expected shape."
+        )
+        try:
+            reshaped_numpy_arr = np.vstack(pd_series.tolist()).reshape(tensor_spec.shape)
```

Comment:
In the above branch (`elif isinstance(pd_series[0], list) and np.isscalar(pd_series[0][0])`), we also convert the data type of the array via

```
            reshaped_numpy_arr = flattened_numpy_arr.reshape(tensor_spec.shape).astype(
                tensor_spec.type
            )
```

Should we do that here too??

Reply:
I think no. Because numpy array includes precise type information, so we don't convert type here, so that in following schema validation we can have strict type check on numpy array column

Reply:
Makes sense - can we add an inline comment explaining why?